core/system/src/mpu/vmpu.c

```diff
@@ -95,6 +95,11 @@ static int vmpu_sanity_checks(void)
     assert( (uint32_t) __uvisor_config.bss_main_end == (SRAM_OFFSET_START +
                                                         UVISOR_SRAM_LENGTH) );
 
+    /* Verify OS private BSS */
+    assert(__uvisor_config.os_priv_end >= __uvisor_config.os_priv_start);
+    assert(__uvisor_config.os_priv_start == __uvisor_config.bss_main_end);
+    assert(__uvisor_config.os_priv_end > __uvisor_config.bss_main_start);
```

Comment:
I'm still not sure this is needed. This section of the sanity checks only check the os private memory section right? In that case I would expect that they check that (i) it is a legit section and (ii) it comes after the end of bss_main.

We give for granted that bss_main is legit as well (it is checked by other checks anyway), so if os_priv_start is == bss_main_end and os_priv_end is >= os_priv_start, then I can trust os_priv_end is > bss_main_start (because I trust bss_main_start < bss_main_end, from other checks)